lib/jobs/ucs-discovery.js

```diff
@@ -186,15 +199,67 @@ function UcsDiscoveryJobFactory(
         });
     };
 
+    UcsDiscoveryJob.prototype.updateRelations = function(nodeId, relations) {
+        // Update existing node with new relations or create one
+        return waterline.nodes.needOneById(nodeId)
+            .then(function(curNode) {
+                relations = _.uniq(relations.concat(curNode.relations), 'relationType');
+                return waterline.nodes.updateOne(
+                    { id: curNode.id },
+                    { relations: relations }
+                );
+            });
+    };
 
-    /**
-     * @function getRoot
-     * @description create obm setting for the node
-     */
-    UcsDiscoveryJob.prototype.createObms = function (nodeId, obm){
-        return waterline.obms.upsertByNode(nodeId, obm)
-            .then(function(n) {
-                return n;
+
+    UcsDiscoveryJob.prototype.createNode = function(data, type) {
+
+        var self = this;
+        var config = Object.assign({}, self.settings);
+
+        assert.string(data.name);
+        assert.string(data.path);
+
+        var id;
+        if (type === 'compute')
+        {
+            assert.array(data.macs);
+            id = data.macs;
+        } else {
+            id = [config.ucs, data.path];
+        }
+
+        var obm = {
+            config: config,
+            service: 'ucs-obm-service'
+        };
+        var node = {
+            type: type,
+            name: data.path,
+            identifiers: id,
+            obm:[obm],
```

Comment:
@geoff-reid This property should be removed.  The obms are added below and live their own collection.